statsmodels/discrete/discrete_model.py

```diff
@@ -747,21 +747,20 @@ def predict(self, params, exog=None, exposure=None, offset=None,
         #TODO: add offset tp
         if exog is None:
             exog = self.exog
-            offset = getattr(self, 'offset', 0)
+        
+        if exposure is None:
             exposure = getattr(self, 'exposure', 0)
-
         else:
-            if exposure is None:
-                exposure = 0
-            else:
-                exposure = np.log(exposure)
-            if offset is None:
-                offset = 0
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = getattr(self, 'offset', 0)
 
+        Xb = np.dot(exog, params[:exog.shape[1]])
```

Comment:
It's your house, so in the interest of closing this out I just renamed Xb --> "fitted" and defined `linpred=fitted+exposure+offset`.  "fitted" lines up nicely with the `fittedvalues` method.